WcaOnRails/app/webpacker/components/OfficersAndBoard/index.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { useMemo, useState } from 'react';
+import {
+  Button, Container, Header, Icon, Popup,
+} from 'semantic-ui-react';
+import { apiV0Urls } from '../../lib/requests/routes.js.erb';
+import { groupTypes } from '../../lib/wca-data.js.erb';
+import I18n from '../../lib/i18n';
+import useLoadedData from '../../lib/hooks/useLoadedData';
+import Loading from '../Requests/Loading';
+import Errored from '../Requests/Errored';
+import UserBadge from '../UserBadge';
+
+// let i18n-tasks know the key is used
+// i18n-tasks-use t('user_roles.status.officers.chair')
+// i18n-tasks-use t('user_roles.status.officers.executive_director')
+// i18n-tasks-use t('user_roles.status.officers.secretary')
+// i18n-tasks-use t('user_roles.status.officers.vice_chair')
+// i18n-tasks-use t('user_roles.status.officers.treasurer')
+
+export default function OfficersAndBoard({ boardEmail }) {
+  const { data: officers, loading: officersLoading, error: officersError } = useLoadedData(
+    apiV0Urls.userRoles.listOfGroupType(groupTypes.officers, 'status', {
+      isActive: true,
+    }),
+  );
+  const { data: board, loading: boardLoading, error: boardError } = useLoadedData(
+    apiV0Urls.userRoles.listOfGroupType(groupTypes.board, 'name', {
+      isActive: true,
+    }),
+  );
+  const [hoveringEmail, setHoveringEmail] = useState(false);
+
+  // The same user can hold multiple officer positions, and it won't be good to show same user
+  // multiple times. Hence, the following function will filter out the duplicates and add the status
+  // in an array for each user.
+  const officerUsers = useMemo(() => {
+    const userIndex = {};
+    return officers?.reduce((accumulator, currentValue) => {
+      const userId = currentValue.user.id;
+      if (userIndex[userId] !== undefined) {
+        accumulator[userIndex[userId]].status.push(currentValue.metadata.status);
+      } else {
+        userIndex[userId] = accumulator.length;
+        accumulator.push({
+          userData: currentValue.user,
+          status: [currentValue.metadata.status],
+        });
+      }
+      return accumulator;
+    }, []);
```

Comment:
I'm not sure if it's worth rewriting, but what if the accumulator were an object with a key per userId, rather then an array with a separate user-index lookup? Then apply `Object.values` to the result.

Reply:
I've thought of that, but I wanted to maintain the sort order, which is why I did like this.

Reply:
I've given a thought again on whether I can group and then sort on client-side. But sorting is not possible in client-side because we need to bring status-sorting logic to client side and it will cause code duplication.

Reply:
It seems like `_.groupBy` preserves the order, so could we use that here?

> [The order of grouped values is determined by the order they occur in collection.](https://lodash.com/docs/#groupBy)

Reply:
Ya thanks for that, done.